In [12]:
import numpy as np
import pandas as pd
from pathlib import Path
import warnings
warnings.filterwarnings('ignore')
from sklearn.linear_model import LogisticRegression
from sklearn.ensemble import RandomForestClassifier

In [6]:
train_df = pd.read_csv(Path('Resources/Generator/2019loans.csv'))
test_df = pd.read_csv(Path('Resources/Generator/2020Q1loans.csv'))

#### Prediction 

Predict forest model classifier would perform better due to classification and numerical data 

In [9]:
# Convert categorical data to numeric and separate target feature for training data
x_train= pd.get_dummies(train_df.drop(columns=['target']))
y_train=train_df['target']

In [10]:
# Convert categorical data to numeric and separate target feature for testing data
x_test= pd.get_dummies(test_df.drop(columns=['target']))
y_test=test_df['target']

In [11]:
# add missing dummy variables to testing set
for col in x_train.columns:
    if col not in x_test.columns:
        x_test[col]=0

In [13]:
# Train the Logistic Regression model on the unscaled data and print the model score (R2squared)
classifier = LogisticRegression()
classifier.fit(x_train, y_train)
classifier.score(x_test, y_test)

0.5089323692045938

In [15]:
# Train a Random Forest Classifier model and print the model score, 
rfc=RandomForestClassifier()
rfc.fit(x_train, y_train)
rfc.score(x_test, y_test)

0.6450446618460229

In [ ]:
# rfc better model for initial data, but neither performed very well.. 

#### Conclusion 

the forest model classifier performed better, confirming prediction rfc 64.5% > 51%. 

#### Prediction 

Predict logistic regression model performs better due to the scaled data between data. 

In [17]:
# Scale the data, -2 to +2 
from sklearn.preprocessing import StandardScaler
scaler = StandardScaler()
scaler.fit(x_train)
#### Prediction 

Predict forest model classifier would perform better due to classification and numerical data 
x_train_scaled=scaler.transform(x_train)
x_test_scaled=scaler.transform(x_test)

In [18]:
# Train the Logistic Regression model on the scaled data and print the model score
classifier2 = LogisticRegression()
classifier2.fit(x_train_scaled, y_train)
classifier2.score(x_test_scaled, y_test)

0.7598894087622289

In [19]:
# Train a Random Forest Classifier model on the scaled data and print the model score
rfc2=RandomForestClassifier()
rfc2.fit(x_train_scaled, y_train)
rfc2.score(x_test_scaled, y_test)

0.6473840918757975

In [ ]:
# logistic regression model performed better in the scaled data 

#### Conclusion 

Logistic Regression model performed better as predicted, 76% > 65%, significantly performing better than the unscaled data. 